runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -3782,9 +3782,26 @@ bool J9::ValuePropagation::isUnreliableSignatureType(
       return false;
 
    int32_t numDims = 0;
+   TR_OpaqueClassBlock *originClass = klass;
    klass = comp()->fej9()->getBaseComponentClass(klass, numDims);
+
    if (!TR::Compiler->cls.isInterfaceClass(comp(), klass))
-      return false;
+      {
+      // If the original class is an array class and it is not a null-restricted array,
+      // we can not trust its type as a nullable array because it can either be a nullable
+      // array class type or a null-restricted array type.
```

Comment:
May I ask you to add a "TODO-VALUETYPE" comment mentioning that once Value Propagation constraints are able to distinguish between nullable and null-restricted arrays that this test can change?